src/main/kotlin/org/opensearch/replication/seqno/RemoteClusterRetentionLeaseHelper.kt

```diff
@@ -179,18 +180,24 @@ class RemoteClusterRetentionLeaseHelper constructor(var followerClusterNameWithU
     /**
      * Remove these once the callers are moved to above APIs
      */
-    public fun addRetentionLease(leaderShardId: ShardId, seqNo: Long,
+    public fun addRetentionLease(leaderIndexShard: IndexShard, leaderShardId: ShardId, seqNo: Long,
                                  followerShardId: ShardId, timeout: Long) {
         val retentionLeaseId = retentionLeaseIdForShard(followerClusterNameWithUUID, followerShardId)
         val request = RetentionLeaseActions.AddRequest(leaderShardId, retentionLeaseId, seqNo, retentionLeaseSource)
         try {
             client.execute(RetentionLeaseActions.Add.INSTANCE, request).actionGet(timeout)
         } catch (e: RetentionLeaseAlreadyExistsException) {
             log.error(e.stackTraceToString())
-            log.info("Renew retention lease as it already exists $retentionLeaseId with $seqNo")
-            // Only one retention lease should exists for the follower shard
-            // Ideally, this should have got cleaned-up
-            renewRetentionLease(leaderShardId, seqNo, followerShardId, timeout)
+            // Only one retention lease should exists for the follower shard and this should've cleared up.
+            // But if the retention lease still exists, renew with RETAIN_ALL can still fail if the last retention lease
+            // was at higher sequence number than the minRetainedSequenceNumber(i.e. RETAIN_ALL).
+            // To get around this, we always renew the existing retention lease with same sequence number to increase the
+            // expiry time.
+            val retainedSequenceNumber = leaderIndexShard.retentionLeases.leases().filter { lease ->
```

Comment:
should we consider the case for expired leases?
Rather than doing this, can we get the seqno from the commitId and use that for renewal if the `RETAIN_ALL`? 
This ensures that, we are not deviating from the underlying assumption for "SYNCING" phase

Reply:
Yes, I think, we should let it fail or remove the lease and add it again. 
In case, we don't follow above, we can risk of loosing the operations b/w the seqno obtained from the commit-id and current seqno(which will be higher) tracked by the current lease. 
Thoughts?

Reply:
Just to simplify this, i've removed the lease renewal as fallback. We'll simply add a lease. If lease exists, we clear the old one and retry.

I've also used `RETAIN_ALL` to keep parity with the existing code..